CallPlayAudio/EventHandler/EventDispatcher.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Communication.CallingServer.Sample.CallPlayAudio
+{
+    using Azure.Communication.CallingServer;
+    using Azure.Messaging;
+    using Newtonsoft.Json;
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading.Tasks;
+
+    public class EventDispatcher
+    {
+        public static readonly EventDispatcher Instance;
+        private readonly ConcurrentDictionary<string, NotificationCallback> NotificationCallback;
+        private object SubscriptionLock = new object();
+
+        static EventDispatcher()
+        {
+            Instance = new EventDispatcher();
+        }
+
+        private EventDispatcher()
+        {
+            NotificationCallback = new ConcurrentDictionary<string, NotificationCallback>(StringComparer.OrdinalIgnoreCase);
+        }
+
+        public bool Subscribe(string eventType, string eventKey, NotificationCallback notificationCallback)
+        {
+            string eventId = BuildEventKey(eventType, eventKey);
+
+            lock (this.SubscriptionLock)
+            {
+                return NotificationCallback.TryAdd(eventId, notificationCallback);
+            }
+        }
+
+        public void Unsubscribe(string eventType, string eventKey)
+        {
+            string eventId = BuildEventKey(eventType, eventKey);
+
+            lock (this.SubscriptionLock)
+            {
+                NotificationCallback.TryRemove(eventId, out _);
+            }
+        }
+
+        public void ProcessNotification(string request)
+        {
+            var callEvent = this.ExtractEvent(request);
+
+            if (callEvent != null)
+            {
+                lock (SubscriptionLock)
+                {
+                    if (NotificationCallback.TryGetValue(GetEventKey(callEvent), out NotificationCallback notificationCallback))
+                    {
+                        if (notificationCallback != null)
+                        {
+                            Task.Run(() =>
+                            {
+                                notificationCallback.Callback.Invoke(callEvent);
+                            });
+                        }
+                    }
+                }
+            }
+        }
+
+        public string GetEventKey(CallingServerEventBase callEventBase)
+        {
+            if (callEventBase is CallConnectionStateChangedEvent)
```

Comment:
you can cast inline. 

`if(callEventBase is CallConnectionStateChangedEvent callConnectionStateChangedEvent)`
`{`
`    var callLegId = callConnectionSateChangedEvent.CallConnectionId;`
`    ...`